api/src/main/java/io/strimzi/api/kafka/model/KafkaJmxOptionsAuthentication.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_DEFAULT)
+@EqualsAndHashCode
+public class KafkaJmxOptionsAuthentication implements UnknownPropertyPreserving, Serializable {
```

Comment:
I think you should rebuild this like other similar classes are. A good inspiration would be https://github.com/strimzi/strimzi-kafka-operator/blob/master/api/src/main/java/io/strimzi/api/kafka/model/authentication/KafkaClientAuthentication.java and  https://github.com/strimzi/strimzi-kafka-operator/blob/master/api/src/main/java/io/strimzi/api/kafka/model/authentication/KafkaClientAuthenticationPlain.java ... you will basically have a class `KafkaJmxOptionsAuthentication` and `KafkaJmxOptionsAuthenticationPassword` and the `type` field would map to `password`. That would make the code as well as the API identical to what we use in other parts of it.

Reply:
Agreed I like that design better!